kube-client/src/lib.rs

```diff
@@ -611,4 +611,137 @@ mod test {
+        // be created before any operations are tested.
+        match pods.create(&Default::default(), &pod).await {
+            Ok(o) => assert_eq!(pod.name_unchecked(), o.name_unchecked()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        let current_ephemeral_containers = pods
+            .get_ephemeral_containers(&pod.name_any())
+            .await?
+            .spec
+            .unwrap()
+            .ephemeral_containers;
+
+        // We expect no ephemeral containers initially, get_ephemeral_containers should
+        // reflect that.
+        assert_eq!(current_ephemeral_containers, None);
+
+        let mut busybox_eph: EphemeralContainer = serde_json::from_value(json!(
+            {
+                "name": "myephemeralcontainer1",
+                "image": "busybox:1.34.1",
+                "command": ["sh", "-c", "sleep 2"],
+            }
+        ))?;
+
+        // Attempt to replace ephemeral containers.
+
+        let patch: Pod = serde_json::from_value(json!({
+            "metadata": { "name": pod_name },
+            "spec":{ "ephemeralContainers": [ busybox_eph ] }
+        }))?;
+
+        let current_containers = pods
+            .replace_ephemeral_containers(&pod_name, &PostParams::default(), &patch)
+            .await?
+            .spec
+            .unwrap()
+            .ephemeral_containers
+            .expect("could find ephemeral container");
+
+        // Note that we can't compare the whole ephemeral containers object, as some fields
+        // are set by the cluster. We therefore compare the fields specified in the patch.
+        assert_eq!(current_containers.len(), 1);
+        assert_eq!(current_containers[0].name, busybox_eph.name);
+        assert_eq!(current_containers[0].image, busybox_eph.image);
+        assert_eq!(current_containers[0].command, busybox_eph.command);
+
+        // Attempt to patch ephemeral containers.
+
+        // The new ephemeral container will have different values from the
+        // first to ensure we can test for its presence.
+        busybox_eph = serde_json::from_value(json!(
+            {
+                "name": "myephemeralcontainer2",
+                "image": "busybox:1.35.0",
+                "command": ["sh", "-c", "sleep 1"],
+            }
+        ))?;
+
+        let patch: Pod =
+            serde_json::from_value(json!({ "spec": { "ephemeralContainers": [ busybox_eph ] }}))?;
+
+        let current_containers = pods
+            .patch_ephemeral_containers(&pod_name, &PatchParams::default(), &Patch::Strategic(patch))
+            .await?
+            .spec
+            .unwrap()
+            .ephemeral_containers
+            .expect("could find ephemeral container");
+
+        // There should only be 2 ephemeral containers at this point,
+        // one from each patch
```

Comment:
don't they go away quickly? your sleep time is very short

Reply:
They still persist in the pod spec :(